### PR TITLE
Delay task execution in AutoCreateAction untill reroute is complete

### PR DESCRIPTION
**Pull request by idegtiarenko**
Related to: https://github.com/elastic/elasticsearch/issues/88647

### CODE REVIEW

**Comment by idegtiarenko** [comment]
There are couple new test failures. I checked and they also fail on the feature branch. I added them to the list: https://github.com/elastic/elasticsearch/issues/86429

File: server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/AllocationActionMultiListenerTests.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class AllocationActionMultiListenerTests extends ESTestCase {
```
Comment (DaveCTurner) [inline server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/AllocationActionMultiListenerTests.java:20]: I think we should have a test that verifies it works correctly (e.g. no lost acks) when accessed from multiple threads too.

**Review by DaveCTurner — CHANGES_REQUESTED** [review]
Implementation looks good, I left a request for another test.

File: rest-api-spec/build.gradle [hunk 2]
```diff
@@ -80,6 +80,8 @@ tasks.named("yamlRestTestV7CompatTransform").configure { task ->
   task.skipTest("search.aggregation/20_terms/numeric profiler", "The profiler results aren't backwards compatible.")
   task.skipTest("migration/10_get_feature_upgrade_status/Get feature upgrade status", "Awaits backport")
   task.skipTest("search/330_fetch_fields/Test disable source", "Error no longer thrown")
+  task.skipTest("indices.split/30_copy_settings/Copy settings during split index", "https://github.com/elastic/elasticsearch/issues/86429")
+  task.skipTest("indices.shrink/30_copy_settings/Copy settings during shrink index", "https://github.com/elastic/elasticsearch/issues/86429")
```
[the remark this answers is not in the document]
Comment (DaveCTurner) [reply rest-api-spec/build.gradle:84]: Was it a `rest-compatibility` job that was failing? This job runs the 7.17 test suite against these 8.x nodes, so muting tests in the 8.x test suite does nothing.

[the remark this answers is not in the document]
Comment (idegtiarenko) [reply rest-api-spec/build.gradle:84]: Yeah, it looks like that:

`./gradlew ':rest-api-spec:yamlRestTestV7CompatTest' --tests "org.elasticsearch.test.rest.ClientYamlTestSuiteIT" -Dtests.method="test {yaml=indices.split/30_copy_settings/Copy settings during split index}"`

**Review by DaveCTurner — APPROVED** [review]
LGTM

**Pull request merged**